### ISSUE TITLE
Repeated calls to readCollectionsFromPath() returns unexpected results when using transforms and .js seeds because of require.cache

### ISSUE THREAD

**Issue by nickpalmer** [pkosiec/mongo-seeding#137]
We are currently using mongo-seeding in our test harness to reseed the database to a common state using `seeder.readCollectionsFromPath()` in a `beforeEach()`;

Recently we added an optional transform for one test which alters the models before running. Of import is that our models are stored in `.js` files.

A test which is run without the transformation after a test with the transform, receives the transformed values, instead of clean values because of `require.cache`.  Note that this is not a problem if you are using .json files since the code uses `readFileSync` and `EJSON.parse` so you get a fresh memory copy each time you call `readCollectionsFromPath()`. So the behavior differs based on the file type you are storing your seeds in.

See: core/src/populator/filesystem.ts:130 => `return require(path);`

This should be changed to use something like https://github.com/sindresorhus/import-fresh to get a fresh copy on each read BEFORE the transformations are applied, so that you get the results you expect with the varying transforms, or the transforms should `cloneDeep()` the model before applying the transforms so the behavior is consistent across file types.

We have implemented a workaround using cloneDeep in our transformer, but it was unexpected that we would need to do so, and cost quite a bit of time to track down.

Our workaround looks something like this:
```
const transform = {
    transformers: [
      SeederCollection => { 
        SeederCollection = cloneDeep(SeederCollection);
        // Further transforms on the SeederCollection
        return SeederCollection;
      }
  ]
};
```

**Comment by pkosiec** [comment]
Hi @nickpalmer,
Thank you for reporting this issue.
I will try to take a look on this and prepare a fix over the weekend. Hopefully next week I will be able to release the new alpha release.

Cheers!

**Comment by pkosiec** [comment]
Hi again @nickpalmer,
This took a bit longer than I expected - sorry for that. I prepared PR with fix: #142, that will be merged soon. There will be just one issue to resolve before next Mongo Seeding release (#131). Thanks for your patience!

**Comment by pkosiec** [comment]
@nickpalmer I've just released new alpha release, which fixes the issue 👉 https://github.com/pkosiec/mongo-seeding/releases/tag/v3.6.0-alpha.0
